rust/examples/metrics-macro.rs

```diff
@@ -0,0 +1,36 @@
+use xaynet_metrics_macro::metrics;
+
+#[cfg(feature = "metrics")]
+use influxdb::{Query, WriteQuery};
+#[cfg(feature = "metrics")]
+use xaynet::metrics;
+
+#[cfg(feature = "metrics")]
+struct Sender;
+
+#[cfg(feature = "metrics")]
+impl Sender {
+    pub fn new() -> Self {
+        Self
+    }
+    pub fn send(&self, query: WriteQuery) {
+        println!("{:?}", query.build())
+    }
+}
+
+#[tokio::main]
+async fn main() {
+    // adds `sender.send(metrics::round::total_number::update(1));`if the feature `metrics` is enabled
+    // `cargo run --example metrics-macro --features metrics` prints `Ok(ValidQuery("state_machine round_total_number=1i 1596543817214696000"))`
+    // however `cargo run --example metrics-macro --features metrics` prints nothing
+    #[cfg(feature = "metrics")]
+    let sender = Sender::new();
+
+    metrics!(
```

Comment:
That was actually my plan to eliminate the dependency of the sender. It is possible but I haven't yet figured out how it works. I took most of the ideas from the `metrics-rs` crate (in particular from the `refactor/metrics-v2` branch). As I understood it, you can solve it with a (global) singleton. I've seen the use of "unsafe" so I've avoided implementing it in this PR😄. If you are interested in their implementation you can take a look at the `metrics-macros/src/lib.rs` and `metrics/src/recorder.rs` file. But I agree, it would be a nice improvement because then we can kick the metrics sender out of the state machine and use the `metrics!` macro anywhere in the code like a `log` macro.